### ISSUE TITLE
Possible bug: Fix ineffective max_norm constraint in EEGNetModel

### ISSUE THREAD

**Issue by GrzegorzSzczepanek** [amrzhd/EEGNet#3]
In the current `EEGNetModel` implementation, the `_apply_max_norm` method is called inside the `__init__` constructor. This means the max-norm constraint is only applied once to the initial random weights and never applied during the actual training process. 

As a result, weights in the Depthwise and Linear layers can grow indefinitely during backpropagation. In EEG classification, this lack of regularization typically causes the model to overfit to high-amplitude artifacts (like EMG/muscle noise) rather than learning true Motor Imagery features, artificially inflating the accuracy.

**Proposed Solution**
The max-norm constraint must be applied after every weight update during training.
1. Remove `_apply_max_norm` calls from `__init__`.
2. Refactor it into a public method `apply_max_norm(self)` wrapped in `torch.no_grad()`.
3. Update the training loops (in `train.py` / notebooks) to call `model.apply_max_norm()` immediately after `optimizer.step()`.

**Implementation Snippet:**
```python
    def apply_max_norm(self):
        """
        Apply Max-Norm Constraint. 
        Must be called in the training loop right after optimizer.step()
        """
        with torch.no_grad():
            # Depthwise layer constraint
            for name, param in self.block2[0].named_parameters():
                if 'weight' in name:
                    param.data = torch.renorm(param.data, p=2, dim=0, maxnorm=self.max_norm1)
            
            # Linear layer constraint
            for name, param in self.fc.named_parameters():
                if 'weight' in name:
                    param.data = torch.renorm(param.data, p=2, dim=0, maxnorm=self.max_norm2)
```
